src/nodes/hypertable_modify.c

```diff
@@ -524,8 +571,48 @@ static CustomPathMethods hypertable_modify_path_methods = {
 	.PlanCustomPath = hypertable_modify_plan_create,
 };
 
+#if PG14_GE
+static bool
+should_chunk_append(Hypertable *ht, PlannerInfo *root, RelOptInfo *rel, Path *path)
+{
+	if (!ts_guc_enable_chunk_append || !ts_guc_enable_optimizations)
+		return false;
+
+	switch (nodeTag(path))
+	{
+		case T_AppendPath:
+			/*
+			 * If there are clauses that have mutable functions, or clauses that reference
+			 * Params this Path might benefit from startup or runtime exclusion
+			 */
+			{
+				AppendPath *append = castNode(AppendPath, path);
+				ListCell *lc;
+
+				/* Don't create ChunkAppend with no children */
+				if (list_length(append->subpaths) == 0)
+					return false;
+
+				foreach (lc, rel->baserestrictinfo)
+				{
+					RestrictInfo *rinfo = (RestrictInfo *) lfirst(lc);
+
+					if (contain_mutable_functions((Node *) rinfo->clause) ||
+						ts_contain_param((Node *) rinfo->clause))
+						return true;
+				}
+				return false;
+				break;
+			}
+		default:
+			return false;
+	}
```

Comment:
```suggestion
	if (nodeTag(path) == T_AppendPath)
	{
		/*
		 * If there are clauses that have mutable functions, or clauses that reference
		 * Params this Path might benefit from startup or runtime exclusion
		 */
		AppendPath *append = castNode(AppendPath, path);
		ListCell *lc;

		/* Don't create ChunkAppend with no children */
		if (list_length(append->subpaths) == 0)
			return false;

		foreach (lc, rel->baserestrictinfo)
		{
			RestrictInfo *rinfo = (RestrictInfo *) lfirst(lc);

			if (contain_mutable_functions((Node *) rinfo->clause) ||
				ts_contain_param((Node *) rinfo->clause))
				return true;
		}
	}
	return false;
```

Do we really need a `switch...case` here?

Reply:
if we ever want to support more then append we need the switch

Reply:
So "if" and "when" it happen then we turn it into a `switch...case` IMHO